pkg/koordlet/runtimehooks/reconciler/reconciler.go

```diff
@@ -41,11 +42,30 @@ const (
 	ContainerLevel ReconcilerLevel = "container"
 )
 
+var (
+	AllQoSClasses = []apiext.QoSClass{apiext.QoSLSE, apiext.QoSLSR, apiext.QoSLS, apiext.QoSBE, apiext.QoSNone}
+	NoneQoSClass  = []apiext.QoSClass{apiext.QoSNone}
+)
+
 // map[string]*cgroupReconciler: key is cgroup filename,
-var globalCgroupReconciler = map[ReconcilerLevel]map[string]*cgroupReconciler{
-	KubeQOSLevel:   {},
-	PodLevel:       {},
-	ContainerLevel: {},
+var globalCgroupReconciler = map[ReconcilerLevel]map[apiext.QoSClass]map[string]*cgroupReconciler{
```

Comment:
this map grows more complex, we need to simplify this.
how about define a struct with []singleReconciler and different kinds of index map(e.g. map[level]singleReconciler).
during the register, we check the reconciler one by one, and during the reconciling, we use the map.

Reply:
done